Source/Core/MapProjection.js

```diff
@@ -65,5 +91,29 @@ define([
      */
     MapProjection.prototype.unproject = DeveloperError.throwInstantiationError;
 
+    var maxcoordRectangleScratch = new Rectangle();
+    var rectangleCenterScratch = new Cartographic();
+    /**
+     * Approximates the extents of a map projection in 2D.
+     *
+     * @function
+     *
+     * @param {MapProjection} mapProjection A map projection from cartographic coordinates to 2D space.
+     * @param {Cartesian3} [result] Optional result parameter.
+     * @private
+     */
+    MapProjection.approximateMaximumCoordinate = function(mapProjection, result) {
+        Check.defined('mapProjection', mapProjection);
+
+        var maxCoord = defaultValue(result, new Cartesian3());
```

Comment:
Don't use `defaultValue` if the second parameter is being created (like `new Cartesian3()`, `{ }`, etc..).  `defaultValue` does not allow for short-circuiting, so this just creates the value and then throws it away, which produces garbage that needs to be collected.  Normally, we do this check at the end and return new `Cartesian3(x,y ,z)` if result isn't defined.

Reply:
Also, is there a good reason to not just make `result` required?  That's always our default position unless there's a good reason for it.